tests/python/test_indices_assert.py

```diff
@@ -0,0 +1,15 @@
+import taichi as ti
+import pytest
+
+@ti.test(debug=True, gdb_trigger=False, exclude=[ti.metal])
+def test_indices_assert():
+    
+    overflow = ti.field(int, (334, 334, 334, 2*10))
```

Comment:
I saw the CI failed because of `std::bad_alloc`. Maybe create a smaller `field`, since the assertion you added doesn't care about the actual size of the field.